illumos-utils/src/zfs.rs

```diff
@@ -307,13 +319,17 @@ impl DatasetProperties {
                     .parse::<u64>()
                     .context("Failed to parse 'used'")?
                     .try_into()?;
+
+                // The values of "quota" and "reservation" can be either "-" or
```

Comment:
This sucks, and is easy to reproduce from any illumos machine.

Try:

```
zfs get quota
```
vs
```
zfs get -o quota
```

With the "parseable" format, values that **were** `none` can become `0`.

Reply:
Just tested this out:

```
$ pfexec zfs set quota=100000 oxi_f37cbbb3-a899-4071-bf0b-318e0d66f61e/backing
-> OK

 $ pfexec zfs set quota=0 oxi_f37cbbb3-a899-4071-bf0b-318e0d66f61e/backing
-> cannot set property for 'oxi_f37cbbb3-a899-4071-bf0b-318e0d66f61e/backing': use 'none' to disable quota/refquota
(fails)

$ pfexec zfs set quota=none oxi_f37cbbb3-a899-4071-bf0b-318e0d66f61e/backing
-> OK
```

So, "no". We cannot set the property to zero, but it is read as zero if it's actually "none".

Reply:
That's good, then! "0 is none" or "you can't set it to 0" are both fine w.r.t. us interpreting 0 as none.